config/src/lib.rs

```diff
@@ -223,9 +223,11 @@ impl ClientConfigBuilder {
 mod tests {
     use super::*;
     use base64::encode;
+    use serial_test::serial;
     use std::net::Ipv4Addr;
 
     #[test]
+    #[serial]
```

Comment:
I don't understand why these tests can't be run in parallel. This is testing config values.
If it's modifying environment variables, can we split that out into its own test so it don't affect others too much?

Reply:
Yes it's because the env var is a shared resource and running unit tests in parallel will have issues. 
I have been googling around and cannot find a better way to deal with shared resource like env var in the unit test. Either we run it consecutively or run it using only 1 thread.